### PR TITLE
feat(onboarding):  add recording on by default step to onboarding flow

### PR DESCRIPTION
**Pull request by alexkim205**
## Context

Add recording setup to the onboarding flow and lets users know it's turned on by default.

https://github.com/PostHog/posthog/issues/13948

### Onboarding Flow V1

https://user-images.githubusercontent.com/13460330/215838625-3fbccc27-253a-4843-82c6-f4b94c57c134.mov

### Onboarding Flow V2

https://user-images.githubusercontent.com/13460330/215834975-936e6dff-e3a9-4b9a-aa1b-53f61389b3a3.mov

## Changes

- [x] Add "Setup session recordings" step to v1 and v2 of onboarding flows.
- [x] Add instrumentation to detect users who interact with session recording step during onboarding flow.
- [x] Add db migration to `team` that changes default for `session_recording_opt_in`, `capture_performance_opt_in`, and `capture_console_log_opt_in` to true.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Locally test with network FF on/off, billing step on/off, v1/v2:

- [x] New project with user seeing ingestion flow for the first time -> on by default
- [x] Existing user with recordings turned off -> settings unchanged

Confirm that migration doesn't affect current teams

### CODE REVIEW

**Comment by raquelmsmith** [comment]
This is kind of a lot of words to shove into the onboarding. I would rather it be more along the lines of a single sentence about session recordings, enable yes or no (default to yes), and then tell them they can configure it more later. But I'm not sure... @kappa90 do you think it matters much?

File: frontend/src/scenes/ingestion/v1/panels/RecordingPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            <p>
+                No further configuration is required to immediately start capturing recordings. Learn more about
+                specific recordings settings in our{' '}
+                <Link to="https://posthog.com/manual/recordings#using-session-recording" target="_blank">
+                    docs
+                </Link>
+                .
+            </p>
+
+            <SessionRecordingSettings inModal />
```
Comment (benjackwhite) [inline frontend/src/scenes/ingestion/v1/panels/RecordingPanel.tsx:26]: Personally I'm not a fan of this. I think just dropping in the Settings here is completely overwhelming for a user. We should make it as concise as possible and enticing to go forward with.

Definitely the case for a custom screen here. Also, we can't use this component anyways as we want to only set things to be enabled when they actually click to continue.

File: frontend/src/scenes/ingestion/v1/panels/RecordingPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <LemonDivider thick dashed className="my-6" />
+            <div>
+                <LemonButton
+                    type="primary"
+                    size="large"
+                    fullWidth
+                    center
+                    className="mb-2"
+                    onClick={() => setVerify(true)}
```
Comment (benjackwhite) [inline frontend/src/scenes/ingestion/v1/panels/RecordingPanel.tsx:35]: We should only actually change the settings when the user clicks continue. If we think about it like a signup form, they can toggle their agreement to terms and conditions checkbox but the actual "agreement" takes place when they press continue.

**Review by benjackwhite — CHANGES_REQUESTED** [review]
I think this needs to be much more concise - something along the lines of this:

Screenshot 2023-02-01 at 09 01 07

**Comment by kappa90** [comment]
Too many checkboxes imho, we want to have just one screen where people can opt-in once for session recordings, and autocapture, they can setup secondary stuff like console logs later?

@raquelmsmith has this on her plate

**Comment by raquelmsmith** [comment]
@alexkim205 I'm happy to take this over if you'd like!

**Comment by alexkim205** [comment]
@raquelmsmith Sure thing! I dont mind at all and you probably have a better idea about how the onboarding flow should feel

Let me know if you have any q's on any changes I've made!

File: posthog/models/team/team.py [hunk 3]
```diff
@@ -130,9 +130,9 @@ class Team(UUIDClassicModel):
     anonymize_ips: models.BooleanField = models.BooleanField(default=False)
     completed_snippet_onboarding: models.BooleanField = models.BooleanField(default=False)
     ingested_event: models.BooleanField = models.BooleanField(default=False)
-    session_recording_opt_in: models.BooleanField = models.BooleanField(default=False)
-    capture_console_log_opt_in: models.BooleanField = models.BooleanField(null=True, blank=True)
-    capture_performance_opt_in: models.BooleanField = models.BooleanField(null=True, blank=True)
+    session_recording_opt_in: models.BooleanField = models.BooleanField(default=True)
```
Comment (raquelmsmith) [inline posthog/models/team/team.py:133]: @alexkim205 do we actually want to default this to `True` for a new team? I thought the idea was to only turn it on for the team _after_ they've seen the ingestion screen and explicitly hit `Continue` with the option set to On.

Comment (alexkim205) [reply posthog/models/team/team.py:133]: Hmm, I was making the assumption here that for all new projects, the user would have gone through the ingestion screen. If we change the default here to `False` we'd have to have extra logic either in the FE or BE that turns these settings on when the onboarding flow is seen—which would be the most accurate way to implement it. Is there the case where a user can manually skip the onboarding flow entirely?

**Comment by neilkakkar** [comment]
Flyby to ensure this behaves as you expect with team caching (as decide is powered by the cache now), specially for existing teams, as new _should be fine_.

File: posthog/migrations/0299_recording_opt_in_default_true.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Generated by Django 3.2.16 on 2023-01-31 17:09
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (ellie) [inline posthog/migrations/0299_recording_opt_in_default_true.py:6]: We will need to ensure that this migration is also followed up with by the new logical replica. cc @guidoiaquinti @hazzadous

File: posthog/models/team/team.py (hunk 3 above)
Comment (raquelmsmith) [reply posthog/models/team/team.py:133]: I think avoiding the migration and defaulting to `False` is the better way to go. That way if someone adds the snippet to their app but gets distracted and never finishes onboarding then we aren't suddenly storing a bunch of session recording data for them. OR say they add the snippet to their site, get distracted, come back a day later, say "no I don't want session recordings", finish onboarding and then go to find that there are actually session recordings from the prior day. I can imagine some side-eye happening there.

I think those two situations are unlikely but I've already handled it on the front-end and we can avoid a migration so I say we do that.

Comment (neilkakkar) [reply posthog/models/team/team.py:133]: That's ideal for any caching logic too, fwiw, as regular updates are handled perfectly well in the cache

File: posthog/migrations/0299_recording_opt_in_default_true.py (hunk 4 above)
Comment (raquelmsmith) [reply posthog/migrations/0299_recording_opt_in_default_true.py:6]: No need, we're gonna nix the migration 👍

**Comment by raquelmsmith** [comment]
Looks like this now:

image

I have a TODO to make autocapture configurable (turn on/off) from the app (instead of just configuring the snippet). However, with other stuff I have on my plate right now that will take me a few days and I don't want to delay shipping this. So I propose we ship this as-is and I can make autocapture settings in the near future.

@kappa90 you okay with that?

File: frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            <div className="my-8">
+                <LemonSwitch
+                    data-attr="opt-in-session-recording-switch"
+                    onChange={(checked) => {
+                        setSessionRecordingsChecked(checked)
+                        updateCurrentTeam({
+                            session_recording_opt_in: checked,
+                            capture_console_log_opt_in: checked,
+                            capture_performance_opt_in: checked,
+                        })
```
Comment (benjackwhite) [inline frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx:46]: I don't love this tbh. I think checking the switch should do nothing, and only the continue code should actually trigger it

**Review by benjackwhite — CHANGES_REQUESTED** [review]
I don't know if this is just my setup but I seem to be getting stuck in a loop state where I can't continue. Got here by creating a new project and going through the flow.

File: frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx (hunk 5 above)
Comment (raquelmsmith) [reply frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx:46]: yeah I think you're right, will change

**Comment by raquelmsmith** [comment]
@benjackwhite I'm not able to reproduce the getting stuck in the ingestion. Maybe @alexkim205 or @kappa90 can try it out locally to see if it happens to them?

**Comment by kappa90** [comment]
I do end up in a loop, trying to figure it out.

**Comment by kappa90** [comment]
Also I'dd add that step after ingesting events, otherwise it's distracting to the integration flow

**Comment by raquelmsmith** [comment]
Oh, I see, if `is_cloud === True` then it doesn't get stuck in a loop, but if it's False it does. Will fix.

**Comment by raquelmsmith** [comment]
I believe I have fixed this 🤞 (the onboarding flow makes me feel like 😵‍💫 sometimes)

File: frontend/src/scenes/ingestion/panels/BillingPanel.tsx [hunk 6]
```diff
@@ -73,6 +73,7 @@ export function BillingPanel(): JSX.Element {
                             center
                             type="tertiary"
                             onClick={() => {
+                                console.log('CURRENT CLICKIND DIS OTHER')
```
Comment (kappa90) [inline frontend/src/scenes/ingestion/panels/BillingPanel.tsx:76]: 👀

File: frontend/src/scenes/ingestion/panels/BillingPanel.tsx [hunk 7]
```diff
@@ -114,6 +115,7 @@ export function BillingPanel(): JSX.Element {
                         center
                         type="tertiary"
                         onClick={() => {
+                            console.log('CURRENT CLICKIND DIS')
```
Comment (kappa90) [inline frontend/src/scenes/ingestion/panels/BillingPanel.tsx:118]: 👀

**Review by kappa90 — APPROVED** [review]
Looking good except for a couple 👀

File: frontend/src/scenes/ingestion/panels/BillingPanel.tsx (hunk 7 above)
Comment (raquelmsmith) [reply frontend/src/scenes/ingestion/panels/BillingPanel.tsx:118]: UGH!

File: frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        <CardContainer
+            nextProps={{ showBilling: showBillingStep }}
+            onContinue={() => {
+                updateCurrentTeam({
+                    session_recording_opt_in: sessionRecordingsChecked,
+                    capture_console_log_opt_in: sessionRecordingsChecked,
+                    capture_performance_opt_in: sessionRecordingsChecked,
+                })
+                if (!showBillingStep) {
+                    console.log('YES HERE')
```
Comment (benjackwhite) [inline frontend/src/scenes/ingestion/panels/SuperpowersPanel.tsx:26]:
```suggestion
```

**Review by benjackwhite — APPROVED** [review]
Approved with one comment to remove a console log 🙌

**Pull request merged**